### PR TITLE
limitrange: skip default limit setting if container's request is larger

### LINKED ISSUE #51430: default limit sometimes acts as max limit
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
I created a `LimitRange` with the following spec:
```
apiVersion: v1
kind: LimitRange
metadata:
  name: mem-limit-range
spec:
  limits:
  - default:
      memory: 512Mi
    defaultRequest:
      memory: 256Mi
    type: Container
```

Now,
- When I create a pod without specifying both requests and limits. It gets request=256Mi and limit=512Mi. Working as expected.
- When I create a pod with limit=x and unspecified request, it gets created with limit=x and request=x. Working as expected.
- When I create a pod with request=x and unspecified limit, it gets created with request=x and limit=512Mi. This is okay when x is less than 512Mi. **But when x is greater than 512Mi, I get an error**:
  ```
  The Pod "blah" is invalid: spec.containers[0].resources.limits: Invalid value: "512Mi": must be greater than or equal to memory request
  ```

**What you expected to happen**:
The default limit should not act as the max limit. It should have allowed me to create a pod with request greater than the default limit. I am not sure what limit value LimitRanger should set in this case, we can either set it equal to request or we can keep it unspecified.

However, if this is expected behavior, I should get a better error message that clearly says that if I am only specifying request, I need to specify it less than the default limit.

**Environment**:
- Kubernetes version (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.4", GitCommit:"793658f2d7ca7f064d2bdf606519f9fe1229c381", GitTreeState:"clean", BuildDate:"2017-08-17T17:03:51Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-07-26T00:12:31Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
```
minikube version: v0.21.0
```

/sig scheduling
/cc @vishh

### PR DESCRIPTION
**Pull request by pacoxu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
default limit sometimes acts as max limit #51430

**Which issue(s) this PR fixes**:
Fixes #51430 

**Special notes for your reviewer**:
Default settings is not enforced as max or min.

The quota in the namespace is by default 500MB limit.
So if a user wants to create a pod with a memory request of 1GB, it should not fail.

- Current Behavior: pod failed with the default 500MB limit, as it is not larger than the requested 1GB.
- Expected: no limit for this pod.
- 

**Does this PR introduce a user-facing change?**:
```release-note
ignore default limit if container's request is larger
```

### CODE REVIEW

File: plugin/pkg/admission/limitranger/admission_test.go [hunk 1]
```diff
@@ -242,7 +244,7 @@ func TestMergePodResourceRequirements(t *testing.T) {
 			t.Errorf("pod %v, expected != actual; %v != %v", pod.Name, expected, actual)
 		}
 	}
-	verifyAnnotation(t, &pod, "LimitRanger plugin set: cpu request for container foo-0; cpu, memory limit for container foo-0")
+	verifyAnnotation(t, &pod, "LimitRanger plugin set: cpu request for container foo-0; cpu limit for container foo-0")
```
Comment (ehashman) [inline plugin/pkg/admission/limitranger/admission_test.go:247]: Do we also want to add a test for the memory case?

Comment (pacoxu) [reply plugin/pkg/admission/limitranger/admission_test.go:247]: fine, will add in testcase later.

Comment (pacoxu) [reply plugin/pkg/admission/limitranger/admission_test.go:247]: Done @ehashman
Thanks.

**Review by ehashman — COMMENTED** [review]
/lgtm
/assign @deads2k

**Comment by pacoxu** [comment]
/cc derekwaynecarr
For comments in #51430

**Comment by pacoxu** [comment]
/assign @deads2k
for approval

**Pull request closed without merging**